components/ledger/pkg/machine/script/compiler/compiler.go

```diff
@@ -78,29 +79,50 @@ func (p *parseVisitor) VisitVariable(c parser.IVariableContext, push bool) (core
 func (p *parseVisitor) VisitExpr(c parser.IExpressionContext, push bool) (core.Type, *core.Address, *CompileError) {
 	switch c := c.(type) {
 	case *parser.ExprAddSubContext:
-		ty, _, err := p.VisitExpr(c.GetLhs(), push)
+		leftOpType, _, err := p.VisitExpr(c.GetLhs(), true)
 		if err != nil {
 			return 0, nil, err
 		}
-		if ty != core.TypeNumber {
-			return 0, nil, LogicError(c, errors.New("tried to do arithmetic with wrong type"))
+		switch leftOpType {
+		case core.TypeNumber:
+		case core.TypeMonetary:
+		default:
+			return 0, nil, LogicError(c, fmt.Errorf(
+				"tried to do an arithmetic operation with an unsupported left-hand side operand type: %s", leftOpType))
 		}
-		ty, _, err = p.VisitExpr(c.GetRhs(), push)
+
+		rightOpType, _, err := p.VisitExpr(c.GetRhs(), true)
 		if err != nil {
 			return 0, nil, err
 		}
-		if ty != core.TypeNumber {
-			return 0, nil, LogicError(c, errors.New("tried to do arithmetic with wrong type"))
+		switch rightOpType {
```

Comment:
This does not handle monetary + number?
Is that ok?

Reply:
It was not in the original design, @altitude do you think it would make sense to allow it?